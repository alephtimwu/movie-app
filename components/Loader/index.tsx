import React from 'react';
import { StyleSheet, View, ActivityIndicator } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center'
  },
  horizontal: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    padding: 10
  }
});

const MyLoader = () => (
  <View style={[styles.container, styles.horizontal]}>
    <ActivityIndicator size="large" color="#fff" />
  </View>
);

export default MyLoader;
