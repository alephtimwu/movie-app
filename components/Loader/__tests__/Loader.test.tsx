import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';
// import { CrpAll } from '__mocks__';
import UseAppContext from '../../../store/index';
import Loader from '../index';

describe('Loader =>', () => {
  let tree = null;

  beforeEach(() => {
    tree = render(<Loader />);
  });

  test('Loader should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  // test('Click on footer back', () => {
  //   const { getAllByText } = tree;
  //   const [back] = getAllByText('Back to Dashboard');
  //   fireEvent.click(back);
  // });
});
