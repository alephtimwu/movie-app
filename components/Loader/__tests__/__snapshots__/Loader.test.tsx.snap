// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader => Loader should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 10,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#fff"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
